AsyncDisplayKit/ASDisplayNode.h

```diff
@@ -306,6 +306,13 @@ extern NSInteger const ASDefaultDrawingPriority;
  */
 @property (nonatomic, readonly, assign) ASSizeRange constrainedSizeForCalculatedLayout;
 
+/**
+ * @abstract Sublcass hook for nodes that are acting as root nodes. This method get's called if one of the subnodes
+ * size is invalidated and may need to result in a different size as the current calculated size.
+ */
+- (void)displayNodeDidInvalidateSizeNewSize:(CGSize)newSize;
```

Comment:
This shouldn't go in ASDisplayNode.h — use Internal or +FrameworkPrivate for now.  Easy to ship the API later.